control-plane/rest/service/src/main.rs

```diff
@@ -241,18 +255,30 @@ async fn main() -> anyhow::Result<()> {
             .expect("Expect to be initialised only once");
     }
 
-    let server =
+    let main_server =
         HttpServer::new(app).bind_rustls_0_23(CliArgs::args().https, get_certificates()?)?;
-    let result = if let Some(http) = CliArgs::args().http {
-        server.bind(http).map_err(anyhow::Error::from)?
+    let main_server = if let Some(http) = CliArgs::args().http {
+        main_server.bind(http).map_err(anyhow::Error::from)?
     } else {
-        server
+        main_server
     }
-    .workers(workers(&CliArgs::args()))
-    .run()
-    .await;
+    .workers(workers(&CliArgs::args()));
+
+    let health_server = HttpServer::new(move || {
```

Comment:
can't we just use the same server?

Reply:
I found this: https://github.com/linkerd/linkerd2/issues/7560
WDYT?

Reply:
Also kubernetes does it with a different API server.

Reply:
A separate server would again consume more memory, which is already a problem.
Also, what happens if the other server is stuck, but the health server is fine. Then we'd be returning potentially incorrect health information?
I'm not sure there's any clear-cut benefit as of now.